lighthouse-core/lib/network-recorder.js

```diff
@@ -353,7 +359,10 @@ class NetworkRecorder extends EventEmitter {
       const initiatorURL = record.initiator.url || (stackFrames[0] && stackFrames[0].url);
       // If we were redirected to this request, our initiator is that redirect, otherwise, it's the
       // initiator provided by the protocol. See https://github.com/GoogleChrome/lighthouse/pull/7352/
-      const initiator = record.redirectSource || recordsByURL.get(initiatorURL);
+      const initiator = record.redirectSource ||
```

Comment:
should probably do that same time check about starting after the initiator here too